src/pyk/kast/manip.py

```diff
@@ -598,8 +598,18 @@ def _rewrites_to_abstractions(_kast: KInner) -> KInner:
 
 
 def anti_unify_with_constraints(
-    constrained_term_1: KInner, constrained_term_2: KInner, implications: bool = False, disjunct: bool = False
+    constrained_term_1: KInner,
+    constrained_term_2: KInner,
+    implications: bool = False,
+    constraint_disjunct: bool = False,
+    abstracted_disjunct: bool = False,
 ) -> KInner:
+    def disjunction_from_substs(substs: list[Subst]) -> KInner:
+        term = substs[0].pred
+        for _i in substs[1:]:
+            term = KApply('_orBool_', [term, _i.pred])
+        return mlEqualsTrue(term)
```

Comment:
It looks like the special case of `len(substs) == 2` is sufficient here:

```suggestion
    def disjunction_from_substs(subst1: Subst, subst2: Subst) -> KInner:
        return mlEqualsTrue(KApply('_orBool_', (subst1.pred, subst2.pred)))
```

Reply:
Also, if you want as compact a constraint as possible, you need to handle the special case of a `subst` being `true`.

Reply:
Simplified as suggested, and also handles the case where `subst.pred` is true.